drivers/pds/lib/utils.go

```diff
@@ -1313,3 +1314,94 @@ func ValidateAllDataServiceVolumes(deployment *pds.ModelsDeployment, dataService
 	return resourceTemp, storageOp, config, nil
 
 }
+
+func GetPodsFromK8sStatefulSet(deployment *pds.ModelsDeployment, namespace string) ([]corev1.Pod, error) {
+	var ss *v1.StatefulSet
+	err = wait.Poll(maxtimeInterval, timeOut, func() (bool, error) {
+		ss, err = k8sApps.GetStatefulSet(deployment.GetClusterResourceName(), namespace)
+		if err != nil {
+			logrus.Warnf("An Error Occured while getting statefulsets %v", err)
+			return false, nil
+		}
+		return true, nil
+	})
+	if err != nil {
+		logrus.Errorf("An Error Occured while getting statefulsets %v", err)
+		return nil, err
+	}
+	pods, err := k8sApps.GetStatefulSetPods(ss)
+	if err != nil {
+		logrus.Errorf("An error occured while getting the pods belonging to this statefulset %v", err)
+		return nil, err
+	}
+	return pods, nil
+}
+
+func GetK8sNodeObjectUsingPodName(nodeName string) (*corev1.Node, error) {
+	nodeObject, err := k8sCore.GetNodeByName(nodeName)
+	if err != nil {
+		logrus.Errorf("Could not get the node object for node %v because %v", nodeName, err)
+		return nil, err
+	}
```

Comment:
Same, just return verbose error, don't print it, we will print it in the initial place where this func is called